lib/internal/process/warning.js

```diff
@@ -89,8 +94,18 @@ function doEmitWarning(warning) {
   process.emit('warning', warning);
 }
 
+const disableWarningsSet = new SafeSet();
+
 function onWarning(warning) {
+  const disableWarningsValues = getOptionValue('--disable-warnings');
+  for (let i = 0; i < disableWarningsValues.length; i++) {
+    disableWarningsSet.add(disableWarningsValues[i]);
```

Comment:
This looks like we repopulate the set on every warning thrown? Shouldn't the set be populated only one, on first use? Since the contents can't change since they were sent in from the flag.

Reply:
good point, fixed!